pkg/statdb/statdb.go

```diff
@@ -65,10 +65,35 @@ func (s *Server) Create(ctx context.Context, createReq *pb.CreateRequest) (resp
 		return nil, err
 	}
 
-	node := createReq.Node
+	var (
+		totalAuditCount    int64
+		auditSuccessCount  int64
+		auditSuccessRatio  float64
+		totalUptimeCount   int64
+		uptimeSuccessCount int64
+		uptimeRatio        float64
+	)
+
+	stats := createReq.Stats
+	if stats != nil {
+		totalAuditCount = stats.AuditCount
+		auditSuccessCount = stats.AuditSuccessCount
+		auditSuccessRatio = float64(auditSuccessCount) / float64(totalAuditCount)
+
+		totalUptimeCount = stats.UptimeCount
+		uptimeSuccessCount = stats.UptimeSuccessCount
+		uptimeRatio = float64(uptimeSuccessCount) / float64(totalUptimeCount)
+
+		// TODO(moby) more specific errors that do not clutter this function
```

Comment:
This is where i would use `errs` error classes. So you could have something like this at the top:
```go
var (
  ErrAuditSuccess = errs.Class("audit success error")
  ErrUptime = errs.Class("...")
  ...
)
```

Then here you can 
```go
if auditSuccessRatio > 1 {
  return nil, ErrAuditSuccess.New("ratio less than 1")
}
```

You may even be able to reduce the conditional statements for all those errors into a switch/case as well.